src/decirest_collection_handler.erl

```diff
@@ -194,11 +194,7 @@ data_prep(Data, PKVal, Children, Req0 = #{path := Path}, State) ->
   SelfUrl = get_self_url(Path, Data, PKVal),
   Req = Req0#{path => SelfUrl},
   ChildUrls = decirest:child_urls_map(Children, Req, State),
-  maps:merge(ChildUrls, maps:remove(self_url_partials, Data#{self_url => SelfUrl}));
-
-data_prep(Data, _, _, Req, _) ->
```

Comment:
Yes, identified by dialyzer.
Given the basic data types used it seems possible, but with the knowledge of how a cowboy-req is defined, it can see that the last case cannot be reached